src/core.rs

```diff
@@ -18,6 +18,7 @@ where
 {
     pub(crate) shape: [usize; N],
     pub(crate) storage: S,
+    pub(crate) len: usize,
```

Comment:
I'd just add this to the storage trait and call down to it if that's the desire, because in the case of the LinearArrayStorage, it's just the length of the array, but in general can likely always be abstracted to a method on the storage implementation.

Reply:
you're right, move it to the storage trait in last commit. len computation occurs at LinearArrayStorage initialisation.